src/ilasm/main.cpp

```diff
@@ -896,11 +916,47 @@ extern "C" int _cdecl wmain(int argc, __in WCHAR **argv)
 #pragma warning(pop)
 #endif
 
-
+#ifndef FEATURE_CORECLR
 HINSTANCE GetModuleInst()
 {
     return (NULL);
 }
+#endif // !FEATURE_CORECLR
 
+#ifdef FEATURE_CORECLR
+int __cdecl main(int argc, char* str[])
+{
+#ifdef FEATURE_PAL
+    g_pszExeFile = str[0];
+    if (0 != PAL_Initialize(argc, str))
+    {
+        fprintf(stderr,"Error: Fail to PAL_Initialize\n");
+        exit(1);
+    }
+#endif
+
+    WCHAR **argv = new WCHAR*[argc];
+    for (int i = 0; i < argc; i++) {
+        int length = MultiByteToWideChar(CP_ACP, 0, str[i], -1, NULL, 0);
+        _ASSERTE(length != 0);
 
+        LPWSTR result = new (nothrow) WCHAR[length];
+        _ASSERTE(result != NULL);
```

Comment:
How about the release build? It would crash if the allocation failed. Could you please change it so that we either print error message and return with an error or use ASSERTE_ALL_BUILDS here? The same probably for the other two asserts.